awkward/array/virtual.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, generator, args=(), kwargs={}, cache=None, persistentkey=None
         self.type = type
         self.nbytes = nbytes
         self.persistvirtual = persistvirtual
-        self._array = None
+        self._array = None if cache is None else self.key
```

Comment:
Was this line a bug? (Like, an oversight when first implementing it? That is, is it assumed that `_array` might be a key elsewhere?)

Reply:
Seems so.. if I follow the `array()` logic, the intention was for `_array` to be always a key if `_cache` existed.